### ISSUE TITLE
Encoding issue ? "Cast string to float is not supported"

### ISSUE THREAD

**Issue by basile753** [perslev/MultiPlanarUNet#34]
Hi there, I'm using your algorythm for a research project at the University of Ottawa and I'm struggling to solve this issue : 

I've encoded my images and label using 3D slicer in the .nii.gz format, choosen the SparseGeneralizedDiceLoss loss function and the sparse_categorical_accuracy metric, and it seems that the program crashes at the start of the training (see Log error below).

I assume this is an encoding mistake, but I can't figure out how to solve that. I tried many different encoding methods in 3Dslicer without success. It might also be a problem with tensorflow but I don't understand because it's the one that fits the requirements.

Thank you so much if someone had an idea to solve that.
Cheers.

Epoch 1/500
2024-11-06 19:00:11.205465: W tensorflow/core/framework/op_kernel.cc:1744] OP_REQUIRES failed at cast_op.cc:124 : Unimplemented: Cast string to float is not supported
 Cast string to float is not supported
         [[node Cast_3 (defined at /usr/lib/python3.7/threading.py:926) ]] [Op:__inference_train_function_6286]

**Comment by christian-igel** [comment]
Hi,
Could you provide a minimal example?
Best wishes,
Christian

**Comment by basile753** [comment]
Hi, thanks a lot for replying so fast !
Sure, here is one .nii manual segmentation (label) file and the corresponding image.
Thank you.
Basile.

[image.nii.gz](https://github.com/user-attachments/files/17664200/image.nii.gz)
[Segmentation.nii.gz](https://github.com/user-attachments/files/17664201/Segmentation.nii.gz)
 
----------------------------------------------------------------------------------------------------------------

Edit : Here also the python functions used to process the data on 3Dslicer : 

# Step 1: Import STL files as segments of a single segmentation

def import_stl_as_segmentation(stl_folder):
    segmentation_node = slicer.mrmlScene.AddNewNodeByClass("vtkMRMLSegmentationNode")
    segmentation_node.CreateDefaultDisplayNodes()  # To visualize the segments
    segment_names = ['Femur', 'Tibia', 'Patella', 'Femur_cartilage', 'Tibia_cartilage_medial', 'Tibia_cartilage_lateral', 'Patella_cartilage', 'Menisc_medial', 'Menisc_lateral']  # List of segment names
    for stl_file in segment_names:
        stl_file += ".stl"
        model_node = slicer.util.loadModel("D:\Antoine\TN10_uOttawa\Data\MRI_Data/1/" + stl_file)
        slicer.modules.segmentations.logic().ImportModelToSegmentationNode(model_node, segmentation_node)
        slicer.mrmlScene.RemoveNode(model_node)
    return segmentation_node

# Step 2: Export the segmentation to .nii.gz 

def export_segmentation_as_nifti(segmentation_node, output_file):
    export_volume_node = slicer.mrmlScene.AddNewNodeByClass("vtkMRMLLabelMapVolumeNode")
    slicer.modules.segmentations.logic().ExportAllSegmentsToLabelmapNode(segmentation_node, export_volume_node)
    slicer.util.saveNode(export_volume_node, output_file)
segmentation_node = import_stl_as_segmentation(stl_folder)
export_segmentation_as_nifti(segmentation_node, output_file)

slicer.app.exit()

**Comment by basile753** [comment]
I actually solved my issue. I figured out that my segmentations didn't have the same shape as my images, I used the "Resample Scalar/Vector/DWI Volume" Module of 3Dlicer to force my segmentation labelmap to have the same dimensions of the images and now the training work.
